### PR TITLE
Add 32bit test in neptun

### PR DESCRIPTION
**Pull request by Hasan6979**
Add test to check rx and tx bytes don't overflow in 32bit systems

### CODE REVIEW

File: neptun/src/noise/mod.rs [hunk 1]
```diff
@@ -1082,4 +1091,35 @@ mod tests {
...
+            let data = if let TunnResult::WriteToNetwork(sent) = data {
+                sent
+            } else {
+                unreachable!();
+            };
+
+            their_tun.decapsulate(None, data, &mut their_dst);
+        }
+
+        let (_, _, rx, _, _) = their_tun.stats();
```
Comment (stalowyjez) [inline neptun/src/noise/mod.rs:1115]: Why won't you merge these two let statements? 😅

File: neptun/src/noise/mod.rs [hunk 2]
```diff
@@ -1082,4 +1091,35 @@ mod tests {
...
+    fn test_long_running_streams() {
+        let (mut my_tun, mut their_tun) = create_two_tuns_and_handshake();
+        let mut my_dst = [0u8; 1500];
+        let mut their_dst = [0u8; 1500];
+
+        let sent_packet_buf = create_large_ipv4_udp_packet();
+
+        for _ in 0..4_000_000 {
+            let data = my_tun.encapsulate(&sent_packet_buf, &mut my_dst);
+            assert!(matches!(data, TunnResult::WriteToNetwork(_)));
```
Comment (stalowyjez) [inline neptun/src/noise/mod.rs:1105]: And this assert looks like it could be instead of `unreachable`

File: .github/workflows/tests.yml [hunk 3]
```diff
@@ -23,6 +23,19 @@ jobs:
...
+    steps:
+    - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
+    - uses: actions-rs/toolchain@b2417cde72dcf67f306c0ae8e0828a81bf0b189f # v1.0.6
+    - run: sudo dpkg --add-architecture i386 && sudo apt-get update
+    - run: sudo apt-get install -y gcc-multilib libc6:i386 libpcap-dev:i386
+    - name: Add 32bit target
+      run: rustup target add i686-unknown-linux-gnu
+    - name: Install hack
+      run: cargo +stable install --git https://github.com/taiki-e/cargo-hack.git cargo-hack --rev c0b517b9eefa27cdaf27cca5f1b186c00ef1af47 --locked
+    - run: cargo hack test --each-feature ${{ matrix.packages }} --target i686-unknown-linux-gnu
```
Comment (stalowyjez) [inline .github/workflows/tests.yml:37]: Maybe we could have some container with this stuff installed instead of doing it every time from scratch?

**Review by stalowyjez — COMMENTED** [review]
Look good, just a few things to consider

File: neptun/src/noise/mod.rs (hunk 1 above)
Comment (stalowyjez) [reply neptun/src/noise/mod.rs:1115]: Oh, nvm
